hack/util.sh

```diff
@@ -96,6 +96,47 @@ function configure_os_server {
 	echo "[INFO] To debug: export KUBECONFIG=$ADMIN_KUBECONFIG"
 }
 
+# check_for_running_openshift is a shortcut for check_for_running "openshift"
+function check_for_running_openshift {
+	check_for_running "openshift"
+}
+# check_for_running_etcd is a shortcut for check_for_running "etcd"
+function check_for_running_etcd {
+	check_for_running "etcd"
+}
+
+# check_for_running displays the parent process for anything that has launched
+# the $1 command.  Existing processes may cause subsequent starts to fail.
+# This function will help identify what is not cleaning up properly.
+function check_for_running {
+	# don't fail for anything, this is just a check
+	set +e
+	echo "[INFO] Scanning for $1 related processes already up"
+	local pids=$(ps h -oppid -C $1)
+
+	# trim any lead space
+	pids="$(echo -e "${pids}" | sed -e 's/^[[:space:]]*//')"
+	if [[ -n "${pids}" ]]; then
+		echo "[WARNING] Existing $1 processes were detected, this may cause unexpected failures : "
+		echo "[WARNING] $1 processes detected : "
+		ps ww -opid,ppid,cmd -C $1
+		show_parent_processes "${pids}"
+	fi
+	set -e
+}
+
+# show_parent_processes is a helper for showing two levels of parent processes.
+function show_parent_processes {
+	echo "[WARNING] Parent processes : "
+	ps ww -opid,ppid,cmd --pid="${1}"
+
+	# try to show one more level.  Wish there was a nicer way to show the forest
```

Comment:
since I really only care about one more level and am formatting it so it doesn't show headers the second time I think I'm ok with it as is unless anyone feels strongly about it.

If not, anything else jumping out or can we shall we get this merged?